src/query/plan/physical.go

```diff
@@ -97,8 +107,8 @@ func (p PhysicalPlan) shiftTime() PhysicalPlan {
 	}
 
 	startShift := maxOffset + maxRange
-	// keeping end the same for now, might optimize later
-	p.TimeSpec.Start = p.TimeSpec.Start.Add(-1 * startShift)
+	alignedShift := startShift - (startShift % p.TimeSpec.Step)
+	p.TimeSpec.Start = p.TimeSpec.Start.Add(-1 * alignedShift)
```

Comment:
This is cropping to the upper bound of the shift, does it need to be the lower bound?

i.e.
```
shift = (max offset + max range)
start - shift
   ↓
|----|----|----|----|
               ↑
              start
```

After adjustment:
```
shift = (max offset + max range)
start - (shift - shift%step)
       ↓
|----|----|----|----|
               ↑
              start
```

Is this desirable or is it more desirable to expand the start further so that the start is "at least" `max offset + max range`?

i.e.
```
shift = (max offset + max range)
start.Add(-1*shift).Truncate(step)
↓      
|----|----|----|----|
               ↑
              start
```
```go
alignedShift := maxOffset + maxRange
p.TimeSpec.Start = p.TimeSpec.Start.Add(-1 * alignedShift).Truncate(p.TimeSpec.Step)
```

Reply:
Ah sorry didn't respond to this, but added a processing step to go back a further StepSize if any alignment was required.